compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -212,6 +212,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 return Ok(());
             }
 
+            Abort => throw_machine_stop!(interpret::Abort),
```

Comment:
I don't think this will work... Miri and const-eval assume that their own error type is the only type ever showing up here. `throw_machine_stop!` is not meant to ever be used in the core interpreter.

Can you make this a machine hook instead?